File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    
    def dfs(n,sm):
        global answer
        if n == len(numbers):
            if sm == target:
                answer += 1
            return
        
        dfs(n+1,sm+numbers[n])
        dfs(n+1,sm-numbers[n])
            
    dfs(0,0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    function dfs(n, sum) {
        if (n === numbers.length) {
          if (sum === target) answer++;
          return;
        }

        dfs(n + 1, sum + numbers[n]);
        dfs(n + 1, sum - numbers[n]);
    }

    dfs(0, 0);

    return answer;
}